Mage/Logger/Writers/ServerWriter.cs

```diff
@@ -1,13 +1,48 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+
+using Newtonsoft.Json.Linq;
 
 
 public class ServerWriter : LogWriter {
-	//private List<string> config;
+	protected Mage mage { get { return Mage.Instance; } }
+
+	private List<string> config;
+
 
 	public ServerWriter(List<string> logLevels) {
-		//config = logLevels;
+		config = logLevels;
+
+		Logger.logEmitter.on("log", HandleLog);
 	}
 
 	public override void Dispose() {
+		Logger.logEmitter.off("log", HandleLog);
+	}
+
+
+	private void HandleLog(object sender, LogEntry logEntry) {
+		if (!config.Contains(logEntry.channel)) {
```

Comment:
Just to be sure, can you null check this ```config``` variable here too
it's not check anywhere else in this file
```csharp
if (config == null || !config.Contains(logEntry.channel)) {
```